server/boards/routes.ts

```diff
@@ -770,4 +771,66 @@ router.get(
   }
 );
 
+/**
+ * @openapi
+ * /boards/{board_id}:
+ *   delete:
+ *     summary: Delete board.
+ *     operationId: deleteBoard
+ *     description: Deletes the specified board.
+ *     tags:
+ *       - /boards/
+ *     security:
+ *       - firebase: []
+ *     parameters:
+ *       - name: board_id
+ *         in: path
+ *         description: The external id of the board to delete.
+ *         required: true
+ *         schema:
+ *           type: string
+ *           format: uuid
+ *         examples:
+ *           existing:
+ *             summary: An existing board
+ *             value: c6d3d10e-8e49-4d73-b28a-9d652b41beec
+ *     responses:
+ *       401:
+ *         description: User is not logged in.
+ *         $ref: "#/components/responses/default401"
+ *       403:
+ *         $ref: "#/components/responses/ensureBoardAccess403"
+ *       404:
+ *         $ref: "#/components/responses/default404"
+ *       500:
+ *         description: Internal Server Error
+ *         $ref: "#/components/responses/default500"
+ *       204:
+ *         description: Board deleted.
+ */
+router.delete(
+  "/:board_id",
+  ensureLoggedIn,
```

Comment:
As thins are right now, this will allow anyone that can see the board to also delete it. I don't know how we want to proceed here: 

1) adding the 501 error (see below) will help cause it means no one can access this in production
2) ...but if we accidentally don't add a real permission check before removing it, it will give anyone permission to hard delete a board

I'm tempted to say this is fine. In a perfect world we'd at least add a test that the board cannot be deleted if process.env is PRODUCTION, but right now I'm inclined to yolo it.